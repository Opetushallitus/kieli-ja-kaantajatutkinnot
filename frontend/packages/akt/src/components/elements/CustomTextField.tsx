import { TextField } from '@mui/material';

import { CustomTextFieldProps } from 'interfaces/components/customTextField';

export const CustomTextField = ({
  error,
  helperText,
  showHelperText,
  multiline,
  ...rest
}: CustomTextFieldProps) => {
  const minRows = multiline ? 5 : undefined;
  const maxRows = multiline ? 15 : undefined;

  return (
    <TextField
      minRows={minRows}
      maxRows={maxRows}
      error={error}
      multiline={multiline}
      helperText={(error || showHelperText) && helperText}
      {...rest}
    />
  );
};
