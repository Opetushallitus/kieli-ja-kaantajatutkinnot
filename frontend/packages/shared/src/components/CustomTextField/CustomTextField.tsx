import { TextField, TextFieldProps } from '@mui/material';

export type CustomTextFieldProps = {
  showHelperText?: boolean;
} & TextFieldProps;

export const CustomTextField = ({
  error,
  helperText,
  showHelperText,
  multiline,
  ...rest
}: CustomTextFieldProps) => {
  const minRows = multiline ? 5 : undefined;
  const maxRows = multiline ? 15 : undefined;

  return (
    <TextField
      minRows={minRows}
      maxRows={maxRows}
      error={error}
      multiline={multiline}
      helperText={(error || showHelperText) && helperText}
      {...rest}
    />
  );
};
