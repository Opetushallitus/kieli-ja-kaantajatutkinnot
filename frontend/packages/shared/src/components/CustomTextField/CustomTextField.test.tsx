import renderer from 'react-test-renderer';

import { CustomTextField } from './CustomTextField';

describe('CustomTextField', () => {
  it('should render correctly with text', () => {
    const tree = renderer
      .create(<CustomTextField label="test label" value="test value" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with error', () => {
    const tree = renderer
      .create(
        <CustomTextField
          label="test label"
          value="wrong test value"
          error={true}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
