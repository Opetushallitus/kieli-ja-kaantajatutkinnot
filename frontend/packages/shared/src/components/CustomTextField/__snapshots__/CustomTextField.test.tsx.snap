// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTextField should render correctly with error 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-error css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={false}
    htmlFor=":r1:"
    id=":r1:-label"
  >
    test label
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={true}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r1:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="wrong test value"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1ftyaf0"
      >
        <span>
          test label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;

exports[`CustomTextField should render correctly with text 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
    data-shrink={true}
    htmlFor=":r0:"
    id=":r0:-label"
  >
    test label
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      id=":r0:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
      value="test value"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-1in441m"
      >
        <span>
          test label
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
